src/lib.rs

```diff
@@ -179,16 +185,20 @@ impl<'a, Writer: Write> Printer<'a, Writer> {
                     }
                 })
                 .collect(),
-            byte_char_table: (0u8..=u8::max_value())
-                .map(|i| {
-                    let byte_char = format!("{}", Byte(i).as_char());
-                    if show_color {
-                        Byte(i).color().paint(byte_char).to_string()
-                    } else {
-                        byte_char
-                    }
+            byte_char_table: show_char_table
+                .then(|| {
```

Comment:
This is failing `cargo clippy` on rust 1.46 (this was set on 2021-11-10, over 6 months ago), if the project is bound to that version I'm happy to whip up an alternative. The run-time cost of this allocation is pretty minimal regardless.

Reply:
I'm fine with bumping the Rust version, if that provides any benefits here.